src/builders/indexable-link-builder.php

```diff
@@ -242,7 +244,14 @@ protected function create_internal_link( $url, $home_url, $indexable, $is_image
 			$target    = $this->indexable_repository->find_by_permalink( $permalink );
 
 			if ( ! $target ) {
-				return $model;
+				// If target indexable cannot be found, create one based on the post's post ID.
```

Comment:
I suggest to make a method `get_post_id` for this to have it simplified. But that's up to you.